buildenv/jenkins/common/build.groovy

```diff
@@ -155,8 +155,19 @@ def checkout_pullrequest() {
 
         // Setup PR_IDs for dependent changes
         for (DEPEND in DEPENDS_ARRAY) {
-            String REPO = DEPEND.substring(DEPEND.indexOf("/") + 1, DEPEND.indexOf("#"));
-            String PR_ID = DEPEND.substring(DEPEND.indexOf("#") + 1, DEPEND.length());
+            switch (DEPEND) {
+                case ~/https:\/\/github\.(ibm\.)?com\/\S+\/\S+\/pull\/[0-9]+/:
+                    def urlList = DEPEND.tokenize('/')
+                    REPO = urlList[3]
+                    PR_ID = urlList[5]
+                    break
+                case ~/\S+\/\S+#[0-9]+/:
```

Comment:
Actually, here we don't want `\S` to match `/` or `#` in the first  segment, so the pattern should be
```
  case ~/[^\s/#]+\/[^\s/#]+#[0-9]+/:
```